app/src/org/commcare/activities/FormEntryActivityUIController.java

```diff
@@ -372,6 +379,10 @@ protected void showNextView() {
     }
 
     private void showNextView(boolean resuming) {
+        if (shouldIgnoreNavigationAction()) {
```

Comment:
Similar to my answer above. `next()` is only used in a small subset of navigation scenarios. It is only called by `onForwardSwipe` and `advance` (which is used for widgets that auto-advance)